src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java

```diff
@@ -212,25 +229,50 @@ public AnalystWorker(Properties config, TransportNetworkCache cache) {
         s3.setRegion(Region.getRegion(Regions.fromName(ec2info.region)));
     }
 
+    /**
+     * Consider shutting down if enough time has passed.
+     * The strategy depends on the fact that we are billed by the hour for EC2 machines.
+     * We only need to consider shutting them down once an hour. Leaving them alive for the rest of the hour provides
+     * immediate compute capacity in case someone starts another job.
+     * So a few minutes before each hour of uptime has elapsed, we check how long the machine has been idle.
+     */
+    public void considerShuttingDown() {
+        long now = System.currentTimeMillis();
+        if (now > nextShutdownCheckTime && autoShutdown) {
+            // Check again exactly one hour later (assumes billing in one-hour increments)
+            nextShutdownCheckTime += 60 * 60 * 1000;
+            if (now > lastSinglePointTime + (SINGLE_KEEPALIVE_MINUTES * 60 * 1000) &&
+                now > lastRegionalTaskTime + (REGIONAL_KEEPALIVE_MINUTES * 60 * 1000)) {
+                LOG.info("Machine has been idle for at least {} minutes (single point) and {} minutes (regional), " +
+                        "shutting down.", SINGLE_KEEPALIVE_MINUTES, REGIONAL_KEEPALIVE_MINUTES);
```

Comment:
Nice.  This will be a welcome improvement from shutting down if the worker happens to be idle at :55 past startup.